Remotion/Data/DomainObjects.UnitTests/QueriesForStandardMapping.xml

```diff
@@ -44,43 +44,43 @@
     </statement>
   </query>
 
-  <query id="BulkUpdateQuery" type="scalar">
+  <query id="BulkUpdateQuery" type="scalar-readwrite">
     <storageGroupType>Remotion.Data.DomainObjects.UnitTests::TestDomain.TestDomainAttribute</storageGroupType>
     <statement>
       update [Order] set [DeliveryDate] = '2006/01/01' where [CustomerID] = @customerID;
       select @@ROWCOUNT;
     </statement>
   </query>
 
-  <query id="OrderQuery" type="collection">
+  <query id="OrderQuery" type="collection-readonly">
     <storageGroupType>Remotion.Data.DomainObjects.UnitTests::TestDomain.TestDomainAttribute</storageGroupType>
     <statement>select [Order].* from [Order] where [CustomerID] = @customerID order by [OrderNo] asc;</statement>
     <collectionType>Remotion.Data.DomainObjects.UnitTests.TestDomain.OrderCollection, Remotion.Data.DomainObjects.UnitTests</collectionType>
   </query>
 
-  <query id="OrderByOfficialQuery" type="collection">
+  <query id="OrderByOfficialQuery" type="collection-readonly">
     <storageGroupType>Remotion.Data.DomainObjects.UnitTests::TestDomain.TestDomainAttribute</storageGroupType>
     <statement>select [Order].* from [Order] where [OfficialID] = @officialID order by [OrderNo] asc;</statement>
     <collectionType>Remotion.Data.DomainObjects.UnitTests.TestDomain.OrderCollection, Remotion.Data.DomainObjects.UnitTests</collectionType>
   </query>
 
-  <query id="CustomerTypeQuery" type="collection">
+  <query id="CustomerTypeQuery" type="collection-readonly">
     <storageGroupType>Remotion.Data.DomainObjects.UnitTests::TestDomain.TestDomainAttribute</storageGroupType>
     <statement>select * from [Company] where [CustomerType] = @customerType order by [Name] asc;</statement>
   </query>
 
-  <query id="QueryWithDuplicates" type="collection">
+  <query id="QueryWithDuplicates" type="collection-readonly">
     <storageGroupType>Remotion.Data.DomainObjects.UnitTests::TestDomain.TestDomainAttribute</storageGroupType>
     <statement>SELECT [Order].* FROM [OrderItem] INNER JOIN [Order] ON [OrderItem].[OrderID] = [Order].[ID] WHERE [Order].[OrderNo] = 1;</statement>
   </query>
 
-  <query id="QueryWithSpecificCollectionType" type="collection">
+  <query id="QueryWithSpecificCollectionType" type="collection-readonly">
     <storageGroupType>Remotion.Data.DomainObjects.UnitTests::TestDomain.TestDomainAttribute</storageGroupType>
     <statement>select * from [Order];</statement>
     <collectionType>Remotion.Data.DomainObjects.UnitTests::TestDomain.SpecificOrderCollection</collectionType>
   </query>
 
-  <query id="QueryWithAllDataTypes" type="collection">
+  <query id="QueryWithAllDataTypes" type="collection-readwrite">
```

Comment:
I dont know what you mean by this

Reply:
this query did not look like it requires a collection-readwrite query type so the change appeared to be simply so we can test a collection-readwrite example. Since the query also had a comment in the body that indicated it tests some error scenario "`<!-- SQL statement with invalid XML character value. -->`" it feels like we would now be using the same query for two different tests (once for an error scenario based on the comment and once for a collection-readwrite query type). This isn't good practice as it causes interdependent tests.